### PR TITLE
Script to check postgres/trino data integrity

### PR DESCRIPTION
**Pull request by lcouzens**
## Jira Ticket

[COST-####](https://issues.redhat.com/browse/COST-####)

## Description

This change will give us a really basic starting point to iterate over active providers/schema to check data completeness between postgres and trino. Essentially a way to check if summary was completed.

This script will collect all schemas based on whether a schema has active providers, it will then build a list of top level PG/Trino tables to query based on schema/active providers, followed by iterating over those tables and checking we have line items for the dates being queried.

What this script do NOT do, is check per provider uuid per table. For example; if you have 3 cloud providers and one of them has not completed summary this script right now will NOT reveal that to you. (could be a future enhancement though) 

## Testing
Run the following for the following results:

python ./dev/scripts/data_integrity.py -g "url-for-gabi-queries" -u "url-for-trino-query" -t "my-token" -c "my.cert" -k "my.key" -s "2023-11-07" -e "2023-11-08" -l

return:
Query dates, Start: 2023-11-07, End: 2023-11-08, Query date step: 5

Context for schema1.
Providers: {'AWS': 1, 'AZURE': 0, 'GCP': 0, 'OCI': 0, 'OCP': 0}
PG Tables to query: {'reporting_awscostentrylineitem_daily_summary'}
Trino Tables to query: {'aws_line_items', 'aws_line_items_daily'}

{'schema1': {'PG': {'reporting_awscostentrylineitem_daily_summary': 'All data complete for table: reporting_awscostentrylineitem_daily_summary'}, 'TRINO': 'PG data complete so skipping trino query'}}

^^^ This will build and return a list of our current active schema

python ./dev/scripts/data_integrity.py -g "url-for-gabi-queries" -u "url-for-trino-query" -t "my-token" -c "my.cert" -k "my.key" -s "2023-11-07" -e "2023-11-08"

return:
['all_Schema']
^^^ This will attempt to validate ALL schema data comparing PG vs Trino data. Any data missing in PG for the start/end dates passed in will trigger the correct relevant trino query and build a list of results for tables that have missing data.

python ./dev/scripts/data_integrity.py -g "url-for-gabi-queries" -u "url-for-trino-query" -t "my-token" -c "my.cert" -k "my.key" -s "2023-11-07" -e "2023-11-08" -x "schema1" "schema2" "schema3"

Return:
['scheam1', 'schema2', 'schema3']

^^^ Run for a specific schema you can also pass in multiple schema

## Notes

In short this gives us a little tool to quickly check if a schema/account has any data missing from summary tables that is present in Trino. Basically if the data is missing from PG but NOT missing from Trino then there is likely a summary issue.

If data is missing from PG and Trino for the same date ranges then its likely not a summary issue.

...

### CODE REVIEW

File: dev/scripts/data_integrity.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+        arg_parser = argparse.ArgumentParser(
+            description="data_integrity: Crawl postgres and trino for missing customer data, format the response as json."  # noqa E501
+        )
+        arg_parser.add_argument(
+            "-g",
+            "--gabi-url",
+            metavar="GABIURL",
+            dest="gabi_url",
+            required=False,
+            help="Gabi instance URL. Can also use env GABI_URL",
```
[the remark this answers is not in the document]
Comment (lcouzens) [reply dev/scripts/data_integrity.py:371]: future enhancement

**Review by maskarb — APPROVED** [review]
lgtm! I think we can make improvements to this script as it gets more use.
